### ISSUE TITLE
emailLog permissions error

### ISSUE THREAD

**Issue by jsuiker** [transistorsoft/cordova-background-geolocation-lt#525]
## Your Environment
* Plugin version: 2.8.5
* Platform: Android
* OS version: 7.0
* Device manufacturer / model: SAMSUNG-SM-G890A
* Cordova version: 7.1.0
* Cordova platform version: 6.3.0
* Plugin config

```
{
  "isMoving": false,
  "enabled": true,
  "odometer": 630289.3,
  "schedulerEnabled": false,
  "trackingMode": "location",
  "distanceFilter": 10,
  "desiredAccuracy": 0,
  "desiredOdometerAccuracy": 100,
  "locationUpdateInterval": 1000,
  "fastestLocationUpdateInterval": 10000,
  "locationTimeout": 60,
  "deferTime": 0,
  "disableElasticity": false,
  "elasticityMultiplier": 1,
  "activityRecognitionInterval": 10000,
  "minimumActivityRecognitionConfidence": 75,
  "triggerActivities": "in_vehicle, on_bicycle, on_foot, running, walking",
  "disableStopDetection": false,
  "stationaryRadius": 25,
  "stopTimeout": 5,
  "stopOnStationary": false,
  "url": "http:\/\/api.XXXXX.dev\/checkin\/christocracy",
  "method": "POST",
  "autoSync": true,
  "autoSyncThreshold": 0,
  "batchSync": false,
  "maxBatchSize": -1,
  "params": {
    "device": {
      "available": true,
      "platform": "Android",
      "version": "7.0",
      "uuid": "4aaae1f025199d4e",
      "cordova": "6.3.0",
      "model": "SAMSUNG-SM-G890A",
      "manufacturer": "samsung",
      "isVirtual": false,
      "serial": "0815f88565d52b04"
    },
    "user": 4
  },
  "headers": {},
  "extras": {},
  "httpRootProperty": "location",
  "locationTemplate": "",
  "geofenceTemplate": "",
  "persist": true,
  "maxDaysToPersist": 1,
  "maxRecordsToPersist": -1,
  "locationsOrderDirection": "ASC",
  "httpTimeout": 60000,
  "geofenceProximityRadius": 1000,
  "geofenceInitialTriggerEntry": true,
  "stopOnTerminate": false,
  "startOnBoot": true,
  "heartbeatInterval": 60,
  "foregroundService": false,
  "notificationPriority": 0,
  "notificationTitle": "Your App Name",
  "notificationText": "Location service activated",
  "notificationColor": "",
  "notificationIcon": "",
  "notificationSmallIcon": "Your App Icon",
  "notificationLargeIcon": "",
  "forceReloadOnLocationChange": false,
  "forceReloadOnMotionChange": false,
  "forceReloadOnGeofence": false,
  "forceReloadOnBoot": false,
  "forceReloadOnHeartbeat": false,
  "forceReloadOnSchedule": false,
  "stopAfterElapsedMinutes": 0,
  "schedule": [],
  "configureUrl": "",
  "configureInterval": -1,
  "debug": true,
  "logLevel": 5,
  "logMaxDays": 3
}
```

## Expected Behavior
The emailLog function should send an email to the address passed as the first parameter.

## Actual Behavior
File Not Found Exception

## Steps to Reproduce
1. Call the emailLog function...

## Context
Retrieve the log file for debugging purposes

## Debug logs
```
D/TSLocationManager: $ emailLog()
D/InputMethodManager: HSI from window - flag : 0 Pid : 6731
E/TSLocationManager: [c.t.l.a.BackgroundGeolocation$f run]
                                                                           ‼️  /storage/emulated/0/background-geolocation.log.gz (Permission denied)
W/System.err: java.io.FileNotFoundException: /storage/emulated/0/background-geolocation.log.gz (Permission denied)
W/System.err:     at java.io.FileOutputStream.open(Native Method)
W/System.err:     at java.io.FileOutputStream.<init>(FileOutputStream.java:221)
W/System.err:     at java.io.FileOutputStream.<init>(FileOutputStream.java:169)
W/System.err:     at com.transistorsoft.locationmanager.adapter.BackgroundGeolocation$f.run(Unknown Source)
W/System.err:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
W/System.err:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
W/System.err:     at java.lang.Thread.run(Thread.java:762)
```

**Comment by christocracy** [comment]
This should be fixed in latest `2.9.0`.

**Comment by jsuiker** [comment]
yes, it works now with 2.9.1
